img/twitter_logo_blue.svg

```diff
@@ -0,0 +1 @@
+<svg id="Logo_FIXED" data-name="Logo — FIXED" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 400 400"><defs><style>.cls-1{fill:none;}.cls-2{fill:#1da1f2;}</style></defs><title>Twitter_Logo_Blue</title><rect class="cls-1" width="400" height="400"/><path class="cls-2" d="M153.62,301.59c94.34,0,145.94-78.16,145.94-145.94,0-2.22,0-4.43-.15-6.63A104.36,104.36,0,0,0,325,122.47a102.38,102.38,0,0,1-29.46,8.07,51.47,51.47,0,0,0,22.55-28.37,102.79,102.79,0,0,1-32.57,12.45,51.34,51.34,0,0,0-87.41,46.78A145.62,145.62,0,0,1,92.4,107.81a51.33,51.33,0,0,0,15.88,68.47A50.91,50.91,0,0,1,85,169.86c0,.21,0,.43,0,.65a51.31,51.31,0,0,0,41.15,50.28,51.21,51.21,0,0,1-23.16.88,51.35,51.35,0,0,0,47.92,35.62,102.92,102.92,0,0,1-63.7,22A104.41,104.41,0,0,1,75,278.55a145.21,145.21,0,0,0,78.62,23"/></svg>
```

Comment:
The convention we used for other SVGs is making them a component. Here are some [action SVGs](https://github.com/morehumaninternet/roar-extension/blob/main/src/popup/components/action-svgs.tsx#L3) for example.

Reply:
@hankeliu2015 following up on this, we tend to include SVGs within our React code so that the images are available immediately when the page renders rather than having the user have to wait for the browser to fetch the image from the specified `src` before it is rendered. In this case, given this is a web extension, that wait time may be very small given the image file is saved on the user's filesystem rather than on our server, but we can be consistent. One area where it matters more is in preferring svg over png where the image will always scale appropriately without loss.

@hankeliu2015 to convert between SVG to JSX, I use [transform.tools](transform.tools). After removing the `height` and `width` attributes it adds and giving my own `className` you have an svg that you can inline in your react code so that the image renders at the same time as the surrounding text and other elements. 

<img width="1434" alt="image" src="https://user-images.githubusercontent.com/6589960/100635895-124cbb80-32ff-11eb-872e-fe830b6cde27.png">

In practice that means we can have a file `src/popup/components/roar-logo.tsx` that looks like this 

```typescript
import * as React from 'react'

export function RoarLogo() {
  return (
    <svg className="roar-logo" viewBox="0 0 128 121" fill="none">
      <path
        d="M44.501 15.888c-18.138 0-19.554-4.221-22.64-4.221-3.085 0-5.861 5.984-5.861 12.357 0 13.013 12.667 17.197 12.667 17.197L44.5 15.888z"
        fill="url(#paint0_linear)"
      />
      <path
        d="M99.333 41.221S112 37.037 112 24.024c0-6.373-2.773-12.357-5.861-12.357-3.086 0-4.502 4.221-22.64 4.221l15.834 25.333z"
        fill="url(#paint1_linear)"
      />
      <path
        d="M99.757 38.555s6.91-2.675 6.91-10.99c0-4.074-1.512-7.898-3.198-7.898-1.682 0-2.456 2.698-12.35 2.698l8.638 16.19z"
        fill="url(#paint2_linear)"
      />
      <path
        d="M36.88 22.365c-9.896 0-10.667-2.698-12.35-2.698-1.682 0-3.197 3.824-3.197 7.898 0 8.318 6.91 10.99 6.91 10.99l8.637-16.19z"
        fill="url(#paint3_linear)"
      />
      <path
        d="M114.667 78.333c0 6.347-12.027 15.067-29.254 15.92-.906.054-1.813.08-2.746.08 0 0-.294.774-.96 1.92-2.054 3.68-7.547 11.414-17.707 11.414-13.333 0-18.667-13.334-18.667-13.334-18.666 0-32-9.333-32-16 0-3.493.294-6.826.747-9.946 1.973-13.574 7.333-22.934 7.333-22.934S30.213 9 64 9c33.787 0 42.587 36.453 42.587 36.453s5.36 9.36 7.333 22.934c.453 3.12.747 6.453.747 9.946z"
        fill="url(#paint4_linear)"
      />
      <path
        d="M93.333 46.333H88c-5.89 0-10.667 4.776-10.667 10.667a5.332 5.332 0 005.334 5.333H88a8 8 0 008-8V49a2.668 2.668 0 00-2.667-2.667z"
        fill="url(#paint5_linear)"
      />
      <path
        d="M77.333 58.333c0-2.208 1.792-4 4-4h8s-2.181 8-8 8c-2.208 0-4-1.792-4-4z"
        fill="url(#paint6_linear)"
      />
      <path
        d="M111.285 56.904a73.744 73.744 0 00-2.365-6.496l-6.683 8.352c-.562.704.174 1.696 1.011 1.363l8.037-3.219z"
        fill="url(#paint7_linear)"
      />
      <path
        d="M114.147 70.16a74.017 74.017 0 00-1.39-7.883l-7.581 9.478c-.637.794.197 1.917 1.141 1.538l7.83-3.133z"
        fill="url(#paint8_linear)"
      />
      <path
        d="M16.715 56.904a73.736 73.736 0 012.365-6.496l6.683 8.352c.562.704-.174 1.696-1.011 1.363l-8.037-3.219z"
        fill="url(#paint9_linear)"
      />
      <path
        d="M34.667 46.333H40c5.89 0 10.667 4.776 10.667 10.667a5.332 5.332 0 01-5.334 5.333H40a8 8 0 01-8-8V49a2.668 2.668 0 012.667-2.667z"
        fill="url(#paint10_linear)"
      />
      <path
        d="M13.853 70.16a73.946 73.946 0 011.39-7.883l7.581 9.478c.637.794-.197 1.917-1.141 1.538l-7.83-3.133z"
        fill="url(#paint11_linear)"
      />
      <path
        d="M46.667 62.333c-5.819 0-8-8-8-8h8c2.208 0 4 1.792 4 4s-1.792 4-4 4z"
        fill="url(#paint12_linear)"
      />
      <path
        d="M50.667 30.333S53.42 26.216 59.728 25H45.333s6.2-5.333 18.667-5.333C76.467 19.667 82.667 25 82.667 25H68.272c6.419 1.355 9.061 5.333 9.061 5.333H50.667z"
        fill="url(#paint13_linear)"
      />
      <path
        d="M56.381 41s1.574-4.117 5.179-5.333h-8.227S56.877 30.333 64 30.333c7.123 0 10.667 5.334 10.667 5.334H66.44C70.107 37.02 71.619 41 71.619 41H56.38z"
        fill="url(#paint14_linear)"
      />
      <path
        d="M82.667 75.667v18.666s-.294.774-.96 1.92c-2.054 3.68-7.547 11.414-17.707 11.414-13.333 0-18.667-13.334-18.667-13.334V75.667h37.334z"
        fill="url(#paint15_linear)"
      />
      <path
        d="M78.667 70.467v-.134h-28c-7.363 0-13.334 5.971-13.334 13.334C37.333 91.029 43.304 97 50.667 97 58.029 97 64 91.03 64 83.667 64 91.027 69.973 97 77.333 97c1.52 0 3.014-.267 4.374-.747a11.87 11.87 0 003.706-2 13.248 13.248 0 005.254-10.586c0-6.904-5.275-12.52-12-13.2z"
        fill="url(#paint16_linear)"
      />
      <path
        d="M53.333 67.667c0 8.25 7.331 10.666 10.667 10.666 3.336 0 10.667-2.416 10.667-10.666H53.333z"
        fill="url(#paint17_linear)"
      />
      <defs>
        <linearGradient
          id="paint0_linear"
          x1={15.8427}
          y1={26.4427}
          x2={33.8427}
          y2={26.4427}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4A81BC" />
          <stop offset={1} stopColor="#164176" />
        </linearGradient>
        <linearGradient
          id="paint1_linear"
          x1={105.608}
          y1={26.4427}
          x2={92.0773}
          y2={26.4427}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4A81BC" />
          <stop offset={1} stopColor="#164176" />
        </linearGradient>
        <linearGradient
          id="paint2_linear"
          x1={98.8933}
          y1={41.0774}
          x2={98.8933}
          y2={16.808}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#FA759E" />
          <stop offset={1} stopColor="#FA759E" />
        </linearGradient>
        <linearGradient
          id="paint3_linear"
          x1={29.1066}
          y1={41.0774}
          x2={29.1066}
          y2={16.808}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#FA759E" />
          <stop offset={1} stopColor="#FA759E" />
        </linearGradient>
        <linearGradient
          id="paint4_linear"
          x1={64}
          y1={-1.352}
          x2={64}
          y2={142.229}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4A81BC" />
          <stop offset={1} stopColor="#164176" />
        </linearGradient>
        <linearGradient
          id="paint5_linear"
          x1={86.6666}
          y1={67.432}
          x2={86.6666}
          y2={41.3786}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#FA759E" />
          <stop offset={1} stopColor="#FA759E" />
        </linearGradient>
        <linearGradient
          id="paint6_linear"
          x1={83.3333}
          y1={39.784}
          x2={83.3333}
          y2={60.6347}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint7_linear"
          x1={101.464}
          y1={56.448}
          x2={112.307}
          y2={53.6826}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint8_linear"
          x1={104.285}
          y1={69.128}
          x2={115.701}
          y2={66.2187}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint9_linear"
          x1={26.536}
          y1={56.448}
          x2={15.6933}
          y2={53.6827}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint10_linear"
          x1={41.3333}
          y1={67.432}
          x2={41.3333}
          y2={41.3786}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#FA759E" />
          <stop offset={1} stopColor="#FA759E" />
        </linearGradient>
        <linearGradient
          id="paint11_linear"
          x1={23.7147}
          y1={69.128}
          x2={12.2987}
          y2={66.2187}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint12_linear"
          x1={44.6667}
          y1={39.784}
          x2={44.6667}
          y2={60.6347}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint13_linear"
          x1={64}
          y1={18.8427}
          x2={64}
          y2={47.6854}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint14_linear"
          x1={64}
          y1={18.8426}
          x2={64}
          y2={47.6853}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
        <linearGradient
          id="paint15_linear"
          x1={64}
          y1={75.432}
          x2={64}
          y2={123.675}
          gradientUnits="userSpaceOnUse"
        >
          <stop offset={0.018} stopColor="#FF1C62" />
          <stop offset={0.541} stopColor="#FA759E" />
          <stop offset={0.819} stopColor="#FA759E" />
        </linearGradient>
        <linearGradient
          id="paint16_linear"
          x1={64}
          y1={106.019}
          x2={64}
          y2={55.9867}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#FA759E" />
          <stop offset={1} stopColor="#FA759E" />
        </linearGradient>
        <linearGradient
          id="paint17_linear"
          x1={64}
          y1={62.96}
          x2={64}
          y2={79.3147}
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#4C4C4C" />
          <stop offset={1} stopColor="#343434" />
        </linearGradient>
      </defs>
    </svg>
  )
}
```

Then the `sign-in-page.scss` can look like this 

```scss
.not-authed {
  min-height: 380px;
  min-width: 476px;
  padding: 10px;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column;
  position: relative;

  button {
    border: none;
    outline: none;
    border-radius: 25px;
    background-color: white;
    color: $twitter-blue;
    padding: 6px 16px 6px 6px;
    font-weight: bold;
    display: flex;
    align-items: center;
  }

  p,
  a {
    color: $human-pink;
    font-size: 18px;
    font-family: 'Roboto', sans-serif;
  }

  svg.roar-logo {
    height: 128px;
  }

  svg.twitter-logo {
    padding: 3px;
    height: 25px;

    path {
      fill: $twitter-blue;
    }
  }
}
```

and the `not-authed.tsx` page can look like this 

```typescript
import * as React from 'react'
import { RoarLogo } from '../components/roar-logo'


function TwitterLogo() {
  return (
    <svg className="twitter-logo" viewBox="0 0 400 400">
      <path
        d="M153.62 301.59c94.34 0 145.94-78.16 145.94-145.94 0-2.22 0-4.43-.15-6.63A104.36 104.36 0 00325 122.47a102.38 102.38 0 01-29.46 8.07 51.47 51.47 0 0022.55-28.37 102.79 102.79 0 01-32.57 12.45 51.34 51.34 0 00-87.41 46.78A145.62 145.62 0 0192.4 107.81a51.33 51.33 0 0015.88 68.47A50.91 50.91 0 0185 169.86v.65a51.31 51.31 0 0041.15 50.28 51.21 51.21 0 01-23.16.88 51.35 51.35 0 0047.92 35.62 102.92 102.92 0 01-63.7 22 104.41 104.41 0 01-12.21-.74 145.21 145.21 0 0078.62 23"
      />
    </svg>
  )
}

export function NotAuthed({ signInWithTwitter }: Pick<Dispatchers<UserAction>, 'signInWithTwitter'>): JSX.Element {
  return (
    <div className="not-authed">
      <RoarLogo />
      <button onClick={signInWithTwitter}>
        <TwitterLogo />
        Log in with Twitter
      </button>
      <p>
        Built with ❤️ by the team at <a target="_blank" href="https://morehumaninternet.org">morehumaninternet.org</a>
      </p>
    </div>
  )
}
```

Reply:
Nice!

Reply:
Made adjustment as instructed